sky/provision/aws/instance.py

```diff
@@ -671,7 +671,9 @@ def terminate_instances(
                                   filters,
                                   included_instances=None,
                                   excluded_instances=None)
-    instances.terminate()
+    instances_list = list(instances)
+    for instance in instances_list:
+        instance.terminate()
```

Comment:
```suggestion
    instances_list = instances.all()
    instances.terminate()
```

https://github.com/boto/boto3/blob/a1754725fb6722041064ccceef8937c3c05a73ff/boto3/resources/collection.py#L186-L205

Reply:
Testing it empirically I found that snippet still runs into the same issue whereas this doesn't:
```suggestion
    instances_list = list(instances)
    instance.terminate()
```
I think it's because the collection that is `instances` or that is returned by `instances.all()` are truly generated on usage, not instantiation.